netbox/core/management/commands/rqworker.py

```diff
@@ -14,6 +16,15 @@ class Command(_Command):
     of only the 'default' queue).
     """
     def handle(self, *args, **options):
+        # Setup system jobs.
+        for job, kwargs in registry['system_jobs'].items():
+            try:
+                interval = kwargs['interval']
+            except KeyError:
+                raise TypeError("System job must specify an interval (in minutes).")
+            logger.debug(f"Scheduling system job {job.name} (interval={interval})")
+            job.enqueue_once(**kwargs)
```

Comment:
From 6b192c9ae851fba587b43aed845f0a46b8230dfc there is a missing `if interval:`.

The reason for this `if` is to selectively ignore job registration, e.g. depending on configuration parameters. An example use case would be not to register the changelog cleanup job if the `CHANGELOG_RETENTION` setting is zero (disabled). If there are other ways to accomplish this (maybe by another decorator before `system_jon()`?) I'm open to that solution as well.

Reply:
Seems like a pretty rare corner case IMO. But in such a case, the job registration could be made conditional. For example:

````python
if CHANGELOG_RETENTION:
    system_job(interval=1440)(HouskeepingJob)
```

Reply:
IMO, I would say the better way to do it is as Jeremy mentioned, or even have the worker itself check to see if it needs to run and if not it can just exit cleanly.